flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/SubtaskExecutionAttemptDetailsInfo.java

```diff
@@ -203,7 +203,8 @@ public static SubtaskExecutionAttemptDetailsInfo create(
         final long now = System.currentTimeMillis();
 
         final TaskManagerLocation location = execution.getAssignedResourceLocation();
-        final String locationString = location == null ? "(unassigned)" : location.getHostname();
+        final String locationString =
+                location == null ? "(unassigned)" : location.getLocationString();
```

Comment:
The `SubtaskExecutionAttemptDetailsInfo#host` should rename, right?

Also, the `SubtaskExecutionAttemptDetailsInfo#FIELD_NAME_HOST` and the front-end should be renamed  as well. However, I'm not sure whether the interface is called by flink users. If yes, the change is incompatible.Should we adding an extra field to store the locationString? 

Anyway, using the `host` to store `locationString` for rest api does't make sense.

Reply:
Other rest api has this problem as well, such as:

- JobVertexTaskManagersInfo.TaskManagersInfo#host
- SubtasksTimesInfo.SubtaskTimeInfo#host
- SubtasksAllAccumulatorsInfo.SubtaskAccumulatorsInfo#host

Reply:
You made a good point. We should by no means rename the field directly as it breaks the backward-compatibility. I propose:
1. Add a new field called "location" (already used in `JobExceptionsInfoWithHistory`) for storing the host + port info.
2. Update the front-end to adopt the new location field.
3. Rename the column name from "Host" to "Location" on the Web UI (previously we reached consensus on using "Host:port", but after a second thought, I think "Location" is more concise)
4. Keep the old "host" field untouched. Notice that the info stored in the old "host" field is inconsistent, sometimes only host is stored, sometimes host + port is stored, I'll leave them as they were with a few comments to keep being compatible.
What do you think @1996fanrui @huwh ?

Reply:
Good catch. It's better to align the name between the UI and the Rest API.

We need a FLIP and discussion in dev ML before this change.

Reply:
As is reminded by others, changing REST API fields would normally require a FLIP. I'll send an email on the discuss thread first and prepare one if needed.